### PR TITLE
Improve organization perspective service URL build

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Related issues
- https://github.com/wso2/product-is/issues/17606

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java [hunk 1]
```diff
@@ -126,4 +143,64 @@ private static String buildErrorDescription(OrganizationManagementConstants.Erro
...
         return errorDescription;
     }
+
+    /**
+     * The relative URL to get the organization.
+     *
+     * @param organizationId The unique identifier of the organization.
+     * @return URI
+     */
+    public static URI organizationGetURL(String organizationId) {
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java:153]: buildOrganizationURL

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java [hunk 2]
```diff
@@ -126,4 +143,64 @@ private static String buildErrorDescription(OrganizationManagementConstants.Erro
...
+        try {
+            url = ServiceURLBuilder.create().addPath(context).build().getRelativePublicURL();
+        } catch (URLBuilderException e) {
+            String errorDescription = "Server encountered an error while building URL for response body.";
+            throw buildInternalServerError(e, errorDescription);
+        }
+        return URI.create(url);
+    }
+
+    private static String getContext(String endpoint) {
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java:173]: Try method overloading

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java]: Fixed by - https://github.com/wso2/identity-api-user/pull/184/commits/3b81ffe597ed52f42cf40d5313553498b37b7813

File: components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.api.user.organization/org.wso2.carbon.identity.rest.api.user.organization.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/organization/v1/util/Util.java]: Fixed by - https://github.com/wso2/identity-api-user/pull/184/commits/3b81ffe597ed52f42cf40d5313553498b37b7813

**Review by dewniMW — APPROVED** [review]

File: components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java [hunk 3]
```diff
@@ -221,13 +231,25 @@ public static URI buildURIForHeader(String endpoint) {
...
+                context = SERVER_API_PATH_COMPONENT + endpoint;
+            } else {
+                context = USER_API_PATH_COMPONENT + endpoint;
+            }
+            if (StringUtils.isNotEmpty(organizationId)) {
+                String tenantDomain = (String) IdentityUtil.threadLocalProperties.get()
+                        .get(OrganizationManagementConstants.ROOT_TENANT_DOMAIN);
+                context = String.format(TENANT_CONTEXT_PATH_COMPONENT, tenantDomain) +
+                        ORGANIZATION_CONTEXT_PATH_COMPONENT + context;
+            }
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java:252]: This logic may not required if this logic is moved to the service URL builder logic in the framework

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java:252]: Fixed by - https://github.com/wso2/identity-api-user/pull/187

**Pull request merged**